openhands/runtime/remote/runtime.py

```diff
@@ -59,13 +59,6 @@ def __init__(
         status_message_callback: Optional[Callable] = None,
     ):
         self.config = config
-        if self.config.sandbox.api_hostname == 'localhost':
-            self.config.sandbox.api_hostname = 'api.all-hands.dev/v0/runtime'
-            logger.info(
-                'Using localhost as the API hostname is not supported in the RemoteRuntime. Please set a proper hostname.\n'
-                'Setting it to default value: api.all-hands.dev/v0/runtime'
-            )
```

Comment:
This had a fallback to an all-hands URL, and we're replacing it with a fallback to localhost. Then some docs need a bit of an update, I think at least this command should now include the var:  https://github.com/All-Hands-AI/OpenHands/tree/main/evaluation/swe_bench#run-inference-on-remoteruntime-experimental

Reply:
I think we *should* set a default for RemoteRuntime though - and throw a warning when `localhost` is in `self.runtime_url` - because `RemoteRuntime` suppose not to be used with `localhost`

Reply:
It should totally be possible to have a "remote runtime" running locally though! At least I plan to do that in dev mode

Reply:
Good point! Then how about we just rename it to `runtime_url` and get rid of the whole `api_hostname` for clarity?